### ISSUE TITLE
add default ProcessingModules

### ISSUE THREAD

**Issue by bendichter** [NeurodataWithoutBorders/nwb-schema#249]
### Problem/Use Case
When I first started using NWB, I did not know how best to use processing modules to organize derived data. After a bit of trial-and-error, I settled on using the NWB core schema module names as a default when they were appropriate. For instance, my `LFP` objects now go in a `ProcessingModule` named `"ecephys"`, my `Position` objects go in a `ProcessingModule` named `"behavior"`, etc. Sometimes you will come across cross-modal data types where these labels do not quite fit, but they do fit for most of the data we will see in these modules. For data that does not in a specific category, users can use the `"misc"` `ProcessingModule` or create a custom module with any name they want. Standardization (even just suggestions) of these `ProcessingModule` names will help users by giving them clear guidance for how to use the processing section, and will promote similar file organization across NWB users.

### Proposal
1) Add **optional** `ProcessingModule`s to the NWB core schema, while leaving intact the ability to create `ProcessingModule`s with custom names. This will provide users with a guide in the schema itself and will be used by matnwb to provide a more direct interface for these default `ProcessingModule`s. This would be similar in spirit to optional columns for `DynamicTables`. Since these processing modules are optional, NWB files missing these modules will still be valid.
2) Add methods to the python API for creating each of these default `ProcessingModule`s. For instance, `nwbfile.modules.behavior` would check if a `ProcessingModule` named `"behavior"` exists. If not, it will create it with an appropriate description, and return that object.

## Checklist

- [x] Have you ensured the feature or change was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues/) ?
- [x] Have you included a brief and descriptive title?
- [x] Have you included a clear description of the problem you are trying to solve?
- [x] Have you included a minimal code snippet that reproduces the issue you are encountering?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?

**Comment by ajtritt** [comment]
Sounds like a good project for someone at the hackathon.

**Comment by bendichter** [comment]
Yeah I agree. I'm mostly interested in getting feedback whether others think this change would be a good idea

**Comment by tjd2002** [comment]
I agree that the flexibility of the format means that it sometimes not obvious how where to start. 

I am not averse to adding some optional processingModule names, but I think the same effect could be achieved by writing more/richer tutorials (see the 'Vignettes' provide with the Seurat package for a nice example of this: https://satijalab.org/seurat/frv). 

Of course, we could also do both: provide optional module names, and then show them in use.

**Comment by bendichter** [comment]
@ajtritt Added to the list of potential projects: https://github.com/NeurodataWithoutBorders/nwb_hackathons/pull/19

I totally agree about documentation. If we develop this feature, we should document how to best use them. So concretely what we need is:

* add default `ProcesssingModule`s to the schema
* add methods to `NWBFile` in pynwb to automatically generate `ProcesssingModule`s when called
* in matnwb, test auto-generation of class attributes for the default `ProcessingModule`s when using the new schema (I think this will work)
* add unit and ui tests for `ProcesssingModule`s
* edit the tutorials to use the default `ProcesssingModule`s

@tjd2002 Those tutorials are great. It would be wonderful to build something like that out as we encounter use-cases.

**Comment by oruebel** [comment]
In addition to updating tutorials, I think this goes in the direction of having documentation for "Best practices" It may very well be that the best practices would just point to the tutorials for details, but I think this would be a good way to document what has been proposed here. E.g., "Best Practice 1: Use the default processing modules when possible" ;-)

**Comment by ajtritt** [comment]
We briefly discussed this in the past in regards to issues raised by @neuromusic when trying to store Ca-imaging data

https://github.com/NeurodataWithoutBorders/pynwb/issues/306#issuecomment-358037285
